airbyte-integrations/connectors/source-recharge/source_recharge/api.py

```diff
@@ -0,0 +1,179 @@
+
+    cursor_field = "created_at"
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        latest_benchmark = latest_record[self.cursor_field]
+        if current_stream_state.get(self.cursor_field):
+            return {self.cursor_field: max(latest_benchmark, current_stream_state[self.cursor_field])}
+        return {self.cursor_field: latest_benchmark}
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None, **kwargs
+    ) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token)
+
+        start_datetime = self._start_date
+        if stream_state.get(self.cursor_field):
+            start_datetime = pendulum.parse(stream_state[self.cursor_field])
+
+        params.update({f"{self.cursor_field}_min": start_datetime.strftime("%Y-%m-%d %H:%M:%S")})
+        return params
+
+
+class Addresses(IncrementalRechargeStream):
+    """
+    Addresses Stream: https://developer.rechargepayments.com/v1-shopify?python#list-addresses
+    """
+
+
+class Charges(IncrementalRechargeStream):
+    """
+    Charges Stream: https://developer.rechargepayments.com/v1-shopify?python#list-charges
+    """
+
+
+class Collections(RechargeStream):
+    """
+    Collections Stream
+    """
+
+
+class Customers(IncrementalRechargeStream):
+    """
+    Customers Stream: https://developer.rechargepayments.com/v1-shopify?python#list-customers
+    """
+
+
+class Discounts(IncrementalRechargeStream):
+    """
+    Discounts Stream: https://developer.rechargepayments.com/v1-shopify?python#list-discounts
+    """
+
+
+class Metafields(RechargeStream):
+    """
+    Metafields Stream: https://developer.rechargepayments.com/v1-shopify?python#list-metafields
+    """
+
+    def read_records(self, stream_slice: Optional[Mapping[str, Any]] = None, **kwargs) -> Iterable[Mapping[str, Any]]:
+        owner_resources = ["customer", "store", "subscription"]
+        for owner in owner_resources:
+            yield from super().read_records(stream_slice={"owner_resource": owner}, **kwargs)
+
+
+class Onetimes(IncrementalRechargeStream):
+    """
+    Onetimes Stream: https://developer.rechargepayments.com/v1-shopify?python#list-onetimes
+    """
+
+
+class Orders(IncrementalRechargeStream):
+    """
+    Orders Stream: https://developer.rechargepayments.com/v1-shopify?python#list-orders
+    """
+
+
+class Products(RechargeStream):
+    """
+    Products Stream: https://developer.rechargepayments.com/v1-shopify?python#list-products
+    """
+
+
+class Shop(RechargeStream):
+    """
+    Shop Stream: https://developer.rechargepayments.com/v1-shopify?python#shop
+    """
+
+    primary_key = ["shop", "store"]
+
+    def get_stream_data(self, response_data: Any) -> List[dict]:
```

Comment:
lets move this logic inside parent class:
* declare data_path property, default return self.name
* if data_path is not none response_data = response_data.get(self.name, [])
* set data_path to None here
WDYT?